### PR TITLE
Add an option in check_Xs to return the number of views, samples and features

### PR DESCRIPTION
**Pull request by pierreablin**
#### What does this implement/fix? Explain your changes.
This adds an optional parameter `return_dimensions` in `mvlearn.utils.check_Xs` so that `check_Xs(Xs, return_dimensions=True)` returns `Xs, n_views, n_samples, n_features`.

This is useful since many estimators need to store these dimensions, and it makes it easier than writing e.g. `n_features = [X.shape[1] for X in Xs]` in each file.

### CODE REVIEW

File: mvlearn/utils/utils.py [hunk 1]
```diff
@@ -38,10 +44,28 @@ def check_Xs(Xs, multiview=False, enforce_views=None, copy=False):
...
+        number of samples, and n_features is a list of length n_views
+        containing the number of features of each view.
+
     Returns
     -------
     Xs_converted : object
         The converted and validated Xs (list of data arrays).
+
+    n_views : int, returned only if return_dimensions is True
+        The number of views in the dataset.
```
Comment (agramfort) [inline mvlearn/utils/utils.py:61]:
```suggestion
    n_views : int
        The number of views in the dataset. Returned only if ``return_dimensions``
        is ``True``.
```

to do below too.

File: mvlearn/utils/utils.py [hunk 2]
```diff
@@ -94,30 +127,60 @@ def check_Xs_y(Xs, y, multiview=False, enforce_views=None):
...
+        Xs_converted, n_views, n_samples, n_features = check_Xs(
+            Xs,
+            multiview=multiview,
+            enforce_views=enforce_views,
+            return_dimensions=True,
+        )
+    else:
+        Xs_converted = check_Xs(
+            Xs, multiview=multiview, enforce_views=enforce_views
+        )
```
Comment (agramfort) [inline mvlearn/utils/utils.py:165]:
```suggestion
    out = check_Xs(
            Xs, multiview=multiview, enforce_views=enforce_views,
            return_dimensions=return_dimensions)
    if return_dimensions:
        Xs_converted, n_views, n_samples, n_features = out
    else:
       Xs_converted = out
```

File: mvlearn/utils/utils.py [hunk 3]
```diff
@@ -94,30 +127,60 @@ def check_Xs_y(Xs, y, multiview=False, enforce_views=None):
...
     _, y_converted = check_X_y(Xs_converted[0], y, allow_nd=False)
 
-    return Xs_converted, y_converted
+    if not return_dimensions:
+        return Xs_converted, y_converted
+    else:
+        return Xs_converted, y_converted, n_views, n_samples, n_features
 
 
 def check_Xs_y_nan_allowed(
```
Comment (agramfort) [inline mvlearn/utils/utils.py:174]: this public function seems unecessary. An allow_nan option in check_Xs_y seems better.

Comment (rflperry) [reply mvlearn/utils/utils.py:174]: Agreed. Address in a new PR addressing #236

File: mvlearn/utils/utils.py [hunk 4]
```diff
@@ -38,10 +44,28 @@ def check_Xs(Xs, multiview=False, enforce_views=None, copy=False):
...
+
+    n_views : int, returned only if return_dimensions is True
+        The number of views in the dataset.
+
+    n_samples : int, returned only if return_dimensions is True
+        The number of samples in the dataset.
+
+    n_views : list, returned only if return_dimensions is True
+        List of lenght n_views containing the number of features for each
+        view.
```
Comment (rflperry) [inline mvlearn/utils/utils.py:68]:
```suggestion
    n_samples : int
        The number of samples in the dataset. Returned only if
        ``return_dimensions`` is ``True``.

    n_features : list
        List of length ``n_views`` containing the number of features in
        each view. Returned only if ``return_dimensions`` is ``True``.
```

File: mvlearn/utils/utils.py [hunk 5]
```diff
@@ -53,17 +77,19 @@ def check_Xs(Xs, multiview=False, enforce_views=None, copy=False):
...
 
     if multiview:
-        if len(Xs) == 1:
+        if n_views == 1:
             msg = "Must provide at least two data matrices"
             raise ValueError(msg)
-        if enforce_views is not None and len(Xs) != enforce_views:
-            msg = "Wrong number of views. Expected {} " \
-                " but found {}".format(enforce_views, len(Xs))
+        if enforce_views is not None and n_views != enforce_views:
```
Comment (rflperry) [inline mvlearn/utils/utils.py:89]: Just a comment but we are supporting only Python 3 so using f strings is fine if you want.

File: mvlearn/utils/utils.py [hunk 6]
```diff
@@ -94,30 +127,60 @@ def check_Xs_y(Xs, y, multiview=False, enforce_views=None):
...
+
+    n_views : int, returned only if return_dimensions is True
+        The number of views in the dataset.
+
+    n_samples : int, returned only if return_dimensions is True
+        The number of samples in the dataset.
+
+    n_views : list, returned only if return_dimensions is True
+        List of lenght n_views containing the number of features for each
+        view.
```
Comment (rflperry) [inline mvlearn/utils/utils.py:153]:
```suggestion
    n_views : int
        The number of views in the dataset. Returned only if ``return_dimensions``
        is ``True``.

    n_samples : int
        The number of samples in the dataset. Returned only if
        ``return_dimensions`` is ``True``.

    n_features : list
        List of length ``n_views`` containing the number of features in
        each view. Returned only if ``return_dimensions`` is ``True``.
```

File: mvlearn/utils/utils.py [hunk 7]
```diff
@@ -94,30 +127,60 @@ def check_Xs_y(Xs, y, multiview=False, enforce_views=None):
...
+        Xs_converted = check_Xs(
+            Xs, multiview=multiview, enforce_views=enforce_views
+        )
     _, y_converted = check_X_y(Xs_converted[0], y, allow_nd=False)
 
-    return Xs_converted, y_converted
+    if not return_dimensions:
+        return Xs_converted, y_converted
+    else:
+        return Xs_converted, y_converted, n_views, n_samples, n_features
```
Comment (rflperry) [inline mvlearn/utils/utils.py:171]:
```suggestion
    if return_dimensions:
        return Xs_converted, y_converted, n_views, n_samples, n_features
    else:
        return Xs_converted, y_converted
```

For consistency and readability.

File: mvlearn/utils/utils.py (hunk 3 above)
Comment (rflperry) [reply mvlearn/utils/utils.py:174]: I would leave this function @pierreablin unchanged, I will address the issue so it's unnecessary to edit it.

Comment (pierreablin) [reply mvlearn/utils/utils.py:174]: ok :)

File: mvlearn/utils/utils.py (hunk 5 above)
Comment (pierreablin) [reply mvlearn/utils/utils.py:89]: sure. I ran black on my code once I wrote it, and it reformatted some things.

**Comment by pierreablin** [comment]
I included your comments, thanks.

**Review by rflperry — APPROVED** [review]

**Review by agramfort — APPROVED** [review]

**Comment by agramfort** [comment]
@rflperry can you give me write access to repo please?

**Comment by agramfort** [comment]
thx @pierreablin

**Pull request merged**